tests/rpc.rs

```diff
@@ -0,0 +1,121 @@
+use monero_harness_rs::Client;
+use spectral::prelude::*;
+use std::time::Duration;
+use testcontainers::clients;
+use tokio::time;
+
+#[tokio::test]
+async fn connect_to_monerod() {
+    let docker = clients::Cli::default();
+    let cli = Client::new_with_random_local_ports(&docker);
+
+    let header = cli
+        .monerod
+        .get_block_header_by_height(0)
+        .await
+        .expect("failed to get block 0");
+
+    assert_eq!(0, header.height);
+}
+
+#[tokio::test]
+async fn miner_is_running_and_producing_blocks() {
+    let docker_client = clients::Cli::default();
```

Comment:
Just a heads-up: `testcontainers` is not built for async contexts as it blocks the thread for every operation. Just be mindful of that because it means you will block the executor thread and not just the single future you are using it in.

Reply:
Oh man that's a big fail, since the whole point of this crate is to enable unit tests that each spin up a container to run in parallel. I guess we still get the advantage of clean/separate state in each container even if the tests end up being run in series.

Reply:
Imho `to run in parallel` is not that important. The library is more about having a clean test environment for our protocols.